app/views/play/level/ControlBarView.coffee

```diff
@@ -182,6 +189,51 @@ module.exports = class ControlBarView extends CocoView
     return if @session.get('state')?.difficulty is @lastDifficulty
     @render()
 
+  onScoresUpdated: (e) ->
+    #return unless @level.get('assessment') in ['open-ended']
+    mainScore = e.scores?[0]
+    if mainScore?.type is 'code-length' and not mainScore.score
+      mainScore = null  # Not counted until complete
+    $scoreboard = @$('#scoreboard').toggle Boolean(mainScore)
+    return unless mainScore
+    utils.replaceText $scoreboard.find('.score-type'), $.i18n.t("leaderboard.#{mainScore.type.replace(/-/g, '_')}")
+    scoreText = @formatScore mainScore.type, mainScore.score, false
+    utils.replaceText $scoreboard.find('.current-score-value'), scoreText
+    bestScore = @topScores[mainScore.type]
+    showBest = bestScore?  # and levelIsOver  # TODO
+    showBest &&= switch mainScore.type
+      when 'time', 'damage-taken' then bestScore < mainScore.score
+      else bestScore > mainScore.score
+    $scoreboard.find('.best-score').toggleClass 'hidden', not showBest
+    if showBest
+      bestScoreText = @formatScore mainScore.type, bestScore, true
+      utils.replaceText $scoreboard.find('.best-score-value'), bestScoreText
+
+  formatScore: (type, score, isBest) ->
+    switch type
+      when 'time', 'survival-time'
+        if isBest
+          score.toFixed(1)
+        else
+          "#{score.toFixed(1)} #{$.i18n.t 'units.seconds'}"
+      else Math.round score
```

Comment:
This logic in general could be put into the vuex store, and this piece of the front end could be replaced with a vue component. If you're feeling like digging in there!

Reply:
You'll have to train me in the ways of the Vue sometime, maybe the next time I have a nice little new component to make.